lib/td/command/job.rb

```diff
@@ -225,24 +226,50 @@ def get_and_show_result(job_id, wait, exclude = false, output = nil, limit = nil
     $stdout.puts "\rUse '-v' option to show detailed messages." + " " * 20 unless verbose
   end
 
-  def wait_job(job, first_call = false)
+  def wait_job(job, first_call = false, wait = nil)
     $stderr.puts "queued..."
 
     cmdout_lines = 0
     stderr_lines = 0
     max_error_counts = JOB_WAIT_MAX_RETRY_COUNT_ON_NETWORK_ERROR
+    timeout_exception_class = Class.new(Exception)
+    deadline = Process.clock_gettime(Process::CLOCK_MONOTONIC, :second) + wait if wait
+    first_exception = nil
 
+    t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :second)
     while first_call || !job.finished?
       first_call = false
       begin
         sleep 2
-        job.update_status!
-      rescue Timeout::Error, SystemCallError, EOFError, SocketError
-        if max_error_counts <= 0
-          raise
+        if wait
+          t = Process.clock_gettime(Process::CLOCK_MONOTONIC, :second)
+          raise timeout_exception_class if deadline < t
```

Comment:
It should be `deadline <= t` instead of `deadline < t`.
Because if `deadline` and `t` are the same, next line will be `Timeout.timeout(0)`. That means never timed out.